### PR TITLE
feat(tools): Add Crowdin PR creator action

### PR DESCRIPTION
**Pull request by RandellDawson**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This PR helps DRY out the existing Crowdin download work flows that @nhcarrigan put together.  The core logic was created in the code he created in the workflow.  I just adapted it to be in a stand-alone action we can use in the repo.  We could potentially move this action to a more global location to use it in other repos if we want to go that route.  I have made it generic enough to accomplish that.

TO DO before before ready to QA:
- [x] Fix some linting issues
- [x] Test it thoroughly

### CODE REVIEW

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:2]:
```suggestion
const githubRoot = require('@actions/github');
```

Renaming this so I can use `github` in a later suggestion (less stuff to rename that way)

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:6]:
```suggestion
    const token = core.getInput('github-token');
    if (!token) {
      core.setFailed('Must provide an API token');
    }
    const branch = core.getInput('branch');
```

- Need to get the token out of the input to pass to a later suggestion.
- `action.yml` and the actual workflow files use `branch` instead of `branch-name`.

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:17]:
```suggestion
    const reviewers = reviewersStr.trim().split(/,\s+/);
    
    const github = githubRoot.getOctokit(token);
```

So the `github` context in the `actions/github-script` I was using is assigned to an Octokit client. Digging through the docs, this is how to get that same client.

Assigned it to `github` so we don't have to modify all the names later in the code.

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:34]:
```suggestion
      head: `${owner}:${branch}`
```

Takes the syntax `nhcarrigan:this-is-my-branch`.

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:63]:
```suggestion
    const prNumber = PR.data.number;
    console.log(
      `https://github.com/freeCodeCamp/freeCodeCamp/pull/${prNumber} created`
    );
```
Gotta flip these or we get an "accessed before initialisation" error

**Review by naomi-lgbt — COMMENTED** [review]
I pulled this in to my actions testing repo and poked at it a bit. These are the changes I had to make to get it to run.

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:25]:
```suggestion
      .catch(() => {
```

File: tools/crowdin/actions/pr-creator/index.js
Comment (nhcarrigan) [inline tools/crowdin/actions/pr-creator/index.js:1]:
```suggestion
/* eslint-disable import/no-unresolved */
/* eslint-disable camelcase */
const core = require('@actions/core');
```

**Comment by RandellDawson** [comment]
Thanks @nhcarrigan for reviewing and making the suggestions.  I think it is ready for others to review.

**Comment by RandellDawson** [comment]
@raisedadead Can we get a second review on this one?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
